mypyc/irbuild/expression.py

```diff
@@ -419,6 +421,18 @@ def transform_op_expr(builder: IRBuilder, expr: OpExpr) -> Value:
     )
 
 
+def try_optimize_int_floor_divide(expr: OpExpr) -> OpExpr:
+    if not isinstance(expr.right, IntExpr):
+        return expr
+    divisor = expr.right.value
+    n = 2
```

Comment:
This isn't super important, but I think it would be cleaner (and faster, though that doesn't really matter) to directly compute the shift instead of testing for it? I think you can do `shift = divisor.bit_length() - 1`  and then check that the shift is correct

Reply:
Done. We don't have a primitive for `bit_length()` so it may actually be slower, but I think that the code looks cleaner if we use it.